src/Contracts/InvoiceInterface.php

```
@@ -80,9 +80,20 @@ public function logo(?string $logoPath): self;
 
     /**
-     * Force a specific total amount that will override the calculated total.
+     * Set an expected total amount for validation purposes.
+     *
+     * When the invoice is rendered, if the expected total differs from the calculated total, an error will be logged.
+     * Set $throw to true to throw an exception instead of just logging.
+     *
+     * @param  float  $amount  The expected total amount
+     * @param  bool  $throw  Whether to throw an exception on mismatch (default: false)
+     */
+    public function expectedTotal(float $amount, bool $throw = false): self;
```

PHPMD (2.15.0)

The method expectedTotal has a boolean flag argument $throw, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)